plugins/linguist-backend/src/processor/LinguistTagsProcessor.ts

```diff
@@ -0,0 +1,236 @@
+      this.bytesThreshold = bytesThreshold;
+    }
+  }
+
+  /**
+   * Given an entity ref, fetches the language breakdown from the Linguist backend HTTP API.
+   * @param entityRef - stringified entity ref
+   * @returns The language breakdown
+   */
+  private async getLanguagesFromLinguistAPI(
+    entityRef: string,
+  ): Promise<string[]> {
+    this.logger.debug(`Fetched languages from linguist API`, {
+      ...this.loggerMeta,
+      entityRef,
+    });
+
+    const baseUrl = this.config.getString('backend.baseUrl');
+    const linguistApi = new URL(`${baseUrl}/api/linguist/entity-languages`);
+    linguistApi.searchParams.append('entityRef', entityRef);
+    const linguistData = await fetch(linguistApi).then(
+      res => res.json() as Promise<Languages>,
+    );
+    if (!linguistData || !linguistData.processedDate) {
+      return [];
+    }
+
+    return linguistData.breakdown
+      .filter(b => b.type === 'programming' && b.bytes > this.bytesThreshold)
+      .map(b => b.name);
+  }
+
+  /**
+   * Cached wrapper around getLanguagesFromLinguistAPI
+   * @param cache - The CatalogProcessorCache
+   * @param entityRef - Stringified entity references
+   *
+   * @returns List of languages
+   */
+  private async getCachedLanguages(
+    cache: CatalogProcessorCache,
+    entityRef: string,
+  ): Promise<string[]> {
+    let cachedData = (await cache.get(entityRef)) as CachedData | undefined;
+    if (!cachedData || this.isExpired(cachedData)) {
+      const languages = await this.getLanguagesFromLinguistAPI(entityRef);
+      cachedData = { languages, cachedTime: Date.now() };
+      await cache.set(entityRef, cachedData);
+    }
+    this.logger.debug(`Fetched cached languages ${cachedData.languages}`, {
+      ...this.loggerMeta,
+      entityRef,
+    });
+    return cachedData.languages;
+  }
+
+  /**
+   * Determines if cached data is expired based on TTL
+   *
+   * @param cachedData - The cached data for this entity
+   * @returns True if data is expired
+   */
+  private isExpired(cachedData: CachedData): boolean {
+    const elapsed = Date.now() - (cachedData.cachedTime || 0);
+    return elapsed > this.cacheTTLMilliseconds;
+  }
+
+  /**
+   * This post-processor will fetch linguist data for a Component and convert the language breakdown
+   * into entity tags which will be appended to the entity.
+   *
+   * @public
+   */
+  async postProcessEntity(
+    entity: Entity,
+    _: any,
+    __: any,
+    cache: CatalogProcessorCache,
+  ): Promise<Entity> {
+    if (!this.entityValidator(entity)) {
+      return entity;
+    }
+    const entityRef = stringifyEntityRef(entity);
+    this.logger.debug(`Processing ${entityRef}`, {
+      ...this.loggerMeta,
+      entityRef,
+    });
+
+    const languages =
+      this.cacheTTLMilliseconds > 0
+        ? await this.getCachedLanguages(cache, entityRef)
+        : await this.getLanguagesFromLinguistAPI(entityRef);
+
+    const tags = (entity.metadata.tags ||= []);
```

Comment:
More a general question, I haven't seen this before but is this the same as `const tags = (entity.metadata.tags || []);`? That `=` after the `||` is throwing me off

Reply:
Yeah, this line is using some trickery.

The `||=` will just set the `tags` property on `entity.metadata` if it doesn't exist already. And then I'm assigning to a local variable so that it's less typing to push data into the tags.

Reply:
Cool, learning things here, thanks! Is there a name for this so I can read up on it a bit more?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR_assignment